backend/src/main/java/ai/verta/modeldb/utils/RdbmsUtils.java

```diff
@@ -1865,22 +1870,34 @@ public static Order getOrderBasedOnSortKey(
 
   private static Predicate getFuzzyUsersQueryPredicate(
       AuthService authService,
+      RoleService roleService,
       CriteriaBuilder builder,
       Root<?> entityRootPath,
       KeyValueQuery requestedPredicate) {
     if (requestedPredicate.getValue().getKindCase().equals(Value.KindCase.STRING_VALUE)) {
       Operator operator = requestedPredicate.getOperator();
       List<UserInfo> userInfoList = getFuzzyUserInfos(authService, requestedPredicate);
       if (userInfoList != null && !userInfoList.isEmpty()) {
-        Expression<String> exp = entityRootPath.get(requestedPredicate.getKey());
-        List<String> vertaIds =
-            userInfoList.stream()
-                .map(authService::getVertaIdFromUserInfo)
-                .collect(Collectors.toList());
+        Set<String> projectIdSet = new HashSet<>();
+        for (UserInfo userInfo : userInfoList) {
+          List<GetResourcesResponseItem> accessibleAllWorkspaceItems =
+              roleService.getResourceItems(
+                  Workspace.newBuilder()
+                      .setId(authService.getWorkspaceIdFromUserInfo(userInfo))
```

Comment:
This is already available and doesn't need an API call, right? It's part of the protos, iirc

Reply:
Yes. worksapceId available in UserInfo proto obj so we are just extracted it in authService.